lib/transport_web/controllers/page_controller.ex

```diff
@@ -17,4 +18,29 @@ defmodule TransportWeb.PageController do
       render conn, "shortlist.html"
   end
 
+  def user_organizations(conn, _) do
+    conn
+    |> get_session(:current_user)
+    |> user_organizations(conn, nil)
+  end
+  def user_organizations(nil, conn, _) do
+    conn
+    |> put_flash(:info, gettext "connection_needed")
+    |> redirect(to: "/login/explanation")
+  end
+  def user_organizations({:error, _}, conn, _) do
```

Comment:
I'd add something like this controller wise :

```elixir
plug :require_authentication

defp require_authentication(conn, _) do
  case assigns do
    %{current_user: current_user} -> {:ok, current_user}
    _ -> conn |> render("500.html")
  end
end
```

Reply:
Or:

```elixir
def user_organizations(conn, _) do
  with {:ok, _} <- require_authentication(conn) do
    #
  else
    render(conn, "401.html")
  end
end

defp require_authentication(conn) do
  case assigns[:current_user] do
    nil -> {:error, :unauthorized}
    current_user -> {:ok, current_user}
  end
end
```

Reply:
I agree, but I'll prefer to do it in another PR.